Could not retrieve keyring: tag byte does not have MSB set

### Helm chart

All

### Helm chart version

All

### Actual behavior (issue description)

Sigstore Helm plugin fails in our release pipeline.

```
Pushed: ghcr.io/contane/charts/foreman:0.3.0
Digest: sha256:5bb958c5[16](https://github.com/contane/charts/actions/runs/9927531924/job/27422602723#step:9:17)360d7260182085442be99028ddab9faffb568bf1af9b79f7a9cb83
Could not retrieve keyring: Error Retrieving Keyring: openpgp: invalid data: tag byte does not have MSB set
Error: plugin "sigstore" exited with error
Error: Process completed with exit code 1.
```

see: https://github.com/contane/charts/actions/runs/9927531924/job/27422602723

### Expected behavior

sigstore should upload our public keyring to Rekor.

### Steps to reproduce

Run a release pipeline.

### Custom configuration

_No response_

### Additional information

There is an issue in the Helm repository (helm/helm#2843). However, the provided solution only outputs the key in binary. I don't think that's a proper solution.

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -42,7 +42,7 @@ jobs:
         env:
           PGP_PRIVATE_KEY: "${{ secrets.PGP_PRIVATE_KEY }}"
           PGP_PASSPHRASE: "${{ secrets.PGP_PASSPHRASE }}"
-          PGP_PUBLIC_KEY: "${{ secrets.PGP_PUBLIC_KEY }}"
+          PGP_PUBLIC_KEY_BINARY: "${{ secrets.PGP_PUBLIC_KEY_BINARY }}"
       - name: Run chart-releaser
         uses: helm/chart-releaser-action@v1.6.0
         with:

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -31,7 +31,9 @@ jobs:
           IFS=""
           echo "$PGP_PRIVATE_KEY" | gpg --dearmor > $HOME/secring.gpg
           echo "$PGP_PASSPHRASE" > $HOME/passphrase.txt
-          echo "$PGP_PUBLIC_KEY" > $HOME/pubring.gpg
+          # Unfortunately, the public key must be in binary format because sigstore does not support ASCII-armored keys.
+          # $PGP_PUBLIC_KEY is base64-encoded, so we decode it to binary and write it to $HOME/pubring.gpg.
+          echo "$PGP_PUBLIC_KEY_BINARY" | base64 --decode > $HOME/pubring.gpg
           
           # Set environment variables in $GITHUB_ENV to make them accessible to subsequent steps in the workflow.
           echo "CR_KEYRING=$HOME/secring.gpg" >> "$GITHUB_ENV"

